jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointCaller.java

```diff
@@ -188,22 +187,28 @@ RegistryErrorException newRegistryErrorException(ResponseException responseExcep
     RegistryErrorExceptionBuilder registryErrorExceptionBuilder =
         new RegistryErrorExceptionBuilder(
             registryEndpointProvider.getActionDescription(), responseException);
-
-    try {
-      ErrorResponseTemplate errorResponse =
-          JsonTemplateMapper.readJson(responseException.getContent(), ErrorResponseTemplate.class);
-      for (ErrorEntryTemplate errorEntry : errorResponse.getErrors()) {
-        registryErrorExceptionBuilder.addReason(errorEntry);
+    if (responseException.getContent() != null) {
+      try {
+        ErrorResponseTemplate errorResponse =
+            JsonTemplateMapper.readJson(
+                responseException.getContent(), ErrorResponseTemplate.class);
+        for (ErrorEntryTemplate errorEntry : errorResponse.getErrors()) {
+          registryErrorExceptionBuilder.addReason(errorEntry);
+        }
+      } catch (IOException ex) {
+        registryErrorExceptionBuilder.addReason(
+            "registry returned error code "
+                + responseException.getStatusCode()
+                + "; possible causes include invalid or wrong reference. Actual error output follows:\n"
+                + responseException.getContent()
+                + "\n");
       }
-    } catch (IOException ex) {
+    } else {
       registryErrorExceptionBuilder.addReason(
           "registry returned error code "
               + responseException.getStatusCode()
-              + "; possible causes include invalid or wrong reference. Actual error output follows:\n"
-              + responseException.getContent()
-              + "\n");
+              + "; but did not return any details; possible causes include invalid or wrong reference, or proxy/firewall/VPN interfering \n");
```

Comment:
nit: minor correction

```suggestion
              + " but did not return any details; possible causes include invalid or wrong reference, or proxy/firewall/VPN interfering \n");
```

Don't forget the test.

Reply:
Done